tests/Integration/Behaviour/Features/Context/Domain/CartRule/SetCartRuleRestrictionsFeatureContext.php

```diff
@@ -73,6 +73,26 @@ public function setRestrictedCartRules(string $cartRuleReference, TableNode $tab
         }
     }
 
+    /**
+     * @When I restrict following carriers for cart rule :cartRuleReference:
+     *
+     * @param string $cartRuleReference
+     * @param TableNode $tableNode
+     *
+     * @return void
+     */
+    public function setRestrictedCarriers(string $cartRuleReference, TableNode $tableNode): void
+    {
+        $command = $this->getRestrictionsCommand($cartRuleReference);
+
+        try {
+            $command->setRestrictedCarrierIds($this->referencesToIds($tableNode->getRowsHash()['restricted carriers']));
+            $this->restrictionCommandsByReference[$cartRuleReference] = $command;
```

Comment:
I was thinking maybe this could be moved into `getRestrictionsCommand`? This would prevent adding this line in all the endpoints no? Or you want to delay it in case the setters trigger an exception?

Reply:
no I don't think getters should ever set anything. Its a trap :smile: Command-query seggregation principle should be applied everywhere (not only for cqrs).

Reply:
It's not exactly the same thing as your getter function here allows you to get a command saved in a field, it's kind of like a cache so it could be responsible for setting it on the first call But anyway it's a detail